src/applets/lock-keys/LockKeysApplet.vala

```diff
@@ -72,6 +88,25 @@ public class LockKeysApplet : Budgie.Applet {
 		}
 	}
 
+	/* Returns true if xdotool is available */
+	protected bool xdotool_installed() {
+		return GLib.FileUtils.test(XDOTOOL, FileTest.IS_EXECUTABLE);
+	}
+
+	/* If xdotool is installed, handle the clicks on the panel icons */
+	protected bool on_panel_widget_clicked(Gdk.EventButton eventbutton, string button) {
+		if (eventbutton.button == 1)
+		{
+			try {
```

Comment:
Should avoid calling if we don't have it in PATH. See https://valadoc.org/glib-2.0/GLib.Environment.find_program_in_path.html, if it is null then it shouldn't try spawning the process.

Reply:
Thanks for feedback. Could I get a clarification please on this point (in case I am misunderstanding.)  The on_panel_widget_clicked callback function doesn't get connected to the button_press_event(s) if xdotool is not found, so without xdotool, this function should never be called.  I didn't add a check here because it felt redundant, but easy enough to add for good measure, if recommended.

Reply:
I don't think it would hurt for us to set up the event listeners anyways and just return the event propagation if we don't have xdotool installed. It wouldn't make a difference to the user and we would get the benefit of the user being able to install xdotool whenever (assuming we don't force it as a rundep) and it would just immediately start working, since `GLib.Environment` would be able to find it in PATH. I don't really expect any meaningful performance regressions resulting from checking it whenever we click it, realistically the user isn't going to be doing it all that often.

Hope that makes sense :)